lib/c-api/src/wasm_c_api/unstable/middlewares/metering.rs

```diff
@@ -0,0 +1,300 @@
+//! Unstable non-standard Wasmer-specific API that contains everything
+//! to create the middleware metering API.
+//!
+//! The metering middleware is used for tracking how many operators
+//! are executed in total and putting a limit on the total number of
+//! operators executed.
+//!
+//! # Example
+//!
+//! ```rust
+//! # use inline_c::assert_c;
+//! # fn main() {
+//! #    (assert_c! {
+//! # #include "tests/wasmer_wasm.h"
+//! #
+//! // Define our “cost function”.
+//! uint64_t cost_function(wasmer_parser_operator_t operator) {
+//!     switch(operator) {
+//!         // `local.get` and `i32.const` cost 1 unit.
+//!         case LocalGet:
+//!         case I32Const:
+//!             return 1;
+//!
+//!         // `i32.add` costs 2 units.
+//!         case I32Add:
+//!             return 2;
+//!
+//!         // The other operations are free.
+//!         default:
+//!             return 0;
+//!     }
+//! }
```

Comment:
Optional: It might be worth using a more realistic cost function for the example?
a) the four Const operations are 0
b) F32Div and F64Div are cost 2
c) everything else is cost 1
?

Reply:
I've taken the example from `examples/metering.rs`. It fits well with the simple WAT module. Thoughts?